src/find_ice_labels.jl

```diff
@@ -1,37 +1,163 @@
-# Type of ref_image_band_2
-__T__ = SubArray{
-    Float64,
-    2,
-    Base.ReinterpretArray{Float64,3,RGB{Float64},Matrix{RGB{Float64}},true},
-    Tuple{Int64,Base.Slice{Base.OneTo{Int64}},Base.Slice{Base.OneTo{Int64}}},
-    false,
-}
+using ImageBinarization: AbstractImageBinarizationAlgorithm, binarize
+"""
+    IceDetectionAlgorithm
+
+Functors to detect ice regions in an image.
+
+Each algorithm `a` with parameters `kwargs...` can be called like:
+- `binarize(image, a(; kwargs...))` 
+- `find_ice(image, a(; kwargs...))`
+- or `a(; kwargs...)(image)`.
 
 """
-    find_reflectance_peaks(reflectance_channel, possible_ice_threshold;)
+abstract type IceDetectionAlgorithm <: AbstractImageBinarizationAlgorithm end
 
-Find histogram peaks in single channels of a reflectance image and return the second greatest peak. If needed, edges can be returned as the first object from `build_histogram`. Similarly, peak values can be returned as the second object from `findmaxima`.
+function (a::IceDetectionAlgorithm)(image::AbstractArray{<:Colorant})
+    return binarize(image, a)
```

Comment:
Very cool that the algorithm can now get passed to `binarize`!

Reply:
Yes, and we get all the other binarization algorithms for free, should we want them.